include/private/fnft__misc.h

```diff
@@ -235,6 +236,66 @@ FNFT_UINT fnft__misc_nextpowerof2(const FNFT_UINT number);
 FNFT_INT fnft__misc_resample(const FNFT_UINT D, const FNFT_REAL eps_t, FNFT_COMPLEX const * const q,
     const FNFT_REAL delta, FNFT_COMPLEX *const q_new);
 
+/**
+ * @brief Multiples two square matrices of size N.
+ *
+ * @ingroup  misc
+ *
+ * Multiples two square matrices U and T of size N. T is replaced by the
+ * result U*T.
+ * @param[in] N Positive integer that is the size of the two matrices.
+ * @param[in] U Pointer to the first element of complex values NxN matrix U.
+ * @param[in,out] T Pointer to the first element of complex values NxN matrix T.
+ * Contains the result U*T on return.
+ */
+static inline void fnft__misc_square_matrix_mult(const FNFT_UINT N, FNFT_COMPLEX * const U,
+        FNFT_COMPLEX *const T){
+    FNFT_UINT  c1, c2, c3;
+    FNFT_COMPLEX TM[32] = { 0 }, sum = 0;
+    for (c1 = 0; c1 < N; c1++) {
+        for (c2 = 0; c2 < N; c2++) {
+            for (c3 = 0; c3 < N; c3++) {
+                sum = sum + U[c1*N+c3]*T[c3*N+c2];
+            }
+            TM[c1*N+c2] = sum;
+            sum = 0;
+        }
+    }
+    memcpy(T,TM,N*N*sizeof(FNFT_COMPLEX));
+    
+    return;
+}
+
+/**
+ * @brief This routine returns the nth degree Legendre polynomial at x.
+ *
+ * @ingroup  misc
+ *
+ * Calculates the the nth degree Legendre polynomial at x using a recursive 
+ * relation (<a href="https://en.wikipedia.org/wiki/Legendre_polynomials#Definition_via_generating_function">Online, Accessed July 2020</a>)
+ * @param[in] n Positive integer that is the order of the Legendre polynomial.
+ * @param[in] x Real scalar value at which the value of the polynomial is to be calculated.
+ * @return Returns the value of nth degree Legendre polynomial at x.
+ */
+static inline FNFT_REAL fnft__misc_legendreP(const FNFT_UINT n, const FNFT_REAL x){
```

Comment:
`misc_legendre_poly` would be more in line with other names

Reply:
Changed.